File: Cc.Core/Models/EventAttachment.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class EventAttachment
    {
        public long Id { get; set; }
        public long EventId { get; set; }
        public string? AttachmentName { get; set; }
        public string? InsertUser { get; set; }
        public DateTime? InsertDate { get; set; }
        public string? UpdateUser { get; set; }
        public DateTime? UpdateDate { get; set; }

        public virtual Event Event { get; set; } = null!;
    }
}

File: Cc.Ui/src/models/eventAttachment.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('eventAttachment')
export class EventAttachment extends Entity {

    @field()
    id: number;
    @field()
    eventId: number;
    @field()
    name: string;
    @field()
    insertUser: string;
    @field()
    insertDate: string;
    @field()
    updateUser: string;
    @field()
    updateDate: string;
    @field()
    file: ArrayBuffer;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): EventAttachment {

        let item = new EventAttachment();
        return item;
    }
}

@resource('eventAttachment')
export class EventAttachmentSearchModel extends Entity implements Pagination {
    
    @field()
    eventId?: number;
    @field()
    name?: string;
    @field()
    file?: ArrayBuffer;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}